src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -476,10 +476,24 @@ public function __clone()
      */
     public function __toString()
     {
+        $protocol = '';
+        try {
+            $protocol = $this->server->get('SERVER_PROTOCOL');
+        } catch (\InvalidArgumentException $mallformedPath) {
+            trigger_error($mallformedPath->getMessage(), E_USER_ERROR);
```

Comment:
Hum, in fact, the dev error handler will throw an exception when this error will be raised. This means end up with the same behavior as before (a fatal error because __toString throws).
In prod, the error message will be a bit better though, but that's a lot of code for a very edge case solution I believe.

Well, looking at the code that throws the exception (in ParameterBag), the only way to trigger it is by using a malformed $path. But here, $path = 'SERVER_PROTOCOL'. There is no way this exception can happen.